src/pages/mine.tsx

```diff
@@ -1,74 +1,67 @@
-            key={item.id}
-            shouldEdit={false}
-            onContentChange={undefined}
-        />
-    ));
+function CardList({ tasks }: { tasks: task[] }) {
+    return (
+        <>
+            {tasks.map((item: task) => (
+                <Card
+                    content={item}
+                    key={item.id}
+                    shouldEdit={false}
+                    onContentChange={undefined}
+                />
+            ))}
+        </>
+    );
 }
 
+const Content = () => {
+    const { data: tasks, error, isLoading } = useGetMineTasksQuery(skipToken);
+    if (isLoading) return <p>Loading...</p>;
+    if (error) return <p>Something went wrong! Please contact admin</p>;
+    if (tasks && tasks.length)
+        return (
+            <div>
+                <CardList tasks={tasks} />
+            </div>
+        );
+    return <p>No Tasks Found</p>;
+};
+
 const Mine: FC = () => {
-    const [tasks, setTasks] = useState<task[]>([]);
-    const { response, error, isLoading, callAPI } = useFetch(
-      MINE_TASKS_URL,
-        {},
-        false
-    );
-    const { state } = useAppContext();
-    const { isLoading: isAuthenticating, isLoggedIn } = state;
-    useEffect(() => {
-        if (isLoggedIn && !Object.keys(response).length) {
-            callAPI();
-            setTasks(response);
-        }
-    }, [isLoggedIn, response]);
+    const { isLoading: isAuthenticating, isLoggedIn } = useAuthenticated();
 
     return (
         <Layout>
             <Head title="Mine" />
             <div className={classNames.container}>
                 <div className={classNames.container}>
-                    {!isAuthenticating &&
-                        (isLoggedIn ? (
-                            isLoading ? (
-                                <p>Loading...</p>
-                            ) : error ? (
-                                <p>
-                                    Something went wrong! Please contact admin
-                                </p>
-                            ) : (
-                                <>
-                                    {tasks.length > 0 ? (
-                                        <div>{CardList(tasks)}</div>
-                                    ) : (
-                                        <p>No Tasks Found</p>
-                                    )}
-                                </>
-                            )
-                        ) : (
-                            <div>
-                                <p>You are not Authorized</p>
-                                <a
-                                    href={LOGIN_URL}
-                                    target="_blank"
-                                    rel="noreferrer"
-                                >
-                                    Click here to Login
-                                </a>
-                            </div>
-                        ))}
+                    {isAuthenticating ? (
```

Comment:
raise another refactor PR for this multi conditional check

Reply:
okay